cartool/main.m

```diff
@@ -134,32 +131,65 @@ void CGImageWriteToFile(CGImageRef image, NSString *path)
 {
     NSMutableArray *images = [[NSMutableArray alloc] initWithCapacity:5];
 
-    for (NSNumber *scaleFactor in @[@1, @2, @3])
-    {
-        CUINamedImage *image = [catalog imageWithName:key scaleFactor:scaleFactor.doubleValue];
+    
+    struct CGPDFDocument *pdfDocument = [catalog pdfDocumentWithName:key];
+    
+    if (pdfDocument) {
```

Comment:
If there's a PDF document, I'm skipping the extraction of the generated raster images and only extracting the PDF